### ISSUE TITLE
[Screen reader] Announcing move mode for top-level blocks without constrained positions

### ISSUE THREAD

**Issue by microbit-robert** [rachel-fenichel/blockly-keyboard-experimentation#757]
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Description

When inserting top-level blocks on to the workspace (or to a lesser extent, putting them into move mode with 'M') where they have no constrained positions, it's not necessarily clear that they are in move mode as this is not announced by screen readers. There's definitely an argument here that this is a learned behaviour, but the difference in readout between blocks with and without constrained positions feels awkward. 

A related issue is when you put any block into move mode (even if it has constrained positions available), confirm it's position, then immediately put it back into move mode again. The fact that it is now in move mode is not announced, probably because the content of the `aria-live` region hasn't changed? Perhaps it needs aria-atomic or needs clearing after a move has been confirmed? Again, less of an issue when the user presses 'M' (unless by accident) as there are also earcons associated with entering and confirming moves.

### Reproduction steps

1. Go to https://google.github.io/blockly-keyboard-experimentation/screenreader/
2. Turn on a screen reader
3. Insert a function block
4. The screen reader output doesn't make it clear that the block is in move mode

### Stack trace

```shell

```

**Comment by cpcallen** [comment]
See also google/blockly#9407.

**Comment by gonfunko** [comment]
Related to RaspberryPiFoundation/blockly#10001

**Comment by BenHenning** [comment]
Very related to https://github.com/RaspberryPiFoundation/blockly/issues/9460.

I wonder if simply ensuring that move mode being entered and exited is properly announced would address the main concern here? Improving move mode for stacks would be nice but probably isn't a pre-requisite to getting the base screen reader behavior in a good place for new stack creation.

**Comment by BenHenning** [comment]
Also related/probably can be tackled at the same time: https://github.com/RaspberryPiFoundation/blockly/issues/9497.

**Comment by maribethb** [comment]
Will be done as part of raspberrypifoundation/blockly#9686
